src/firefox/index.js

```diff
@@ -249,9 +278,45 @@ export async function useProfile(
     app,
     configureThisProfile = configureProfile,
     customPrefs = {},
+    createProfileFinder = defaultCreateProfileFinder,
   }: UseProfileParams = {},
 ): Promise<FirefoxProfile> {
-  const profile = new FirefoxProfile({destinationDirectory: profilePath});
+  let profile;
+  let defaultProfilePath = '';
+  let defaultDevProfilePath = '';
+  const finder = createProfileFinder();
+  try {
+    const dirExists = await isDirectory(profilePath);
+    if (await finder.hasProfileName('default')) {
+      defaultProfilePath = await finder.getPath('default');
+    }
+    if (await finder.hasProfileName('dev-edition-default')) {
+      defaultDevProfilePath = await finder.getPath('dev-edition-default');
+    }
+    if (dirExists) {
+      log.debug(`Using profile directory from "${profilePath}"`);
+      if (profilePath === defaultProfilePath ||
+        profilePath === defaultDevProfilePath) {
+        throw new WebExtError(
```

Comment:
By looking at the coverage reports, this line doesn't seem to be currently executed by any of the current tests.